server.go

```diff
@@ -141,6 +154,28 @@ func (s *Server) loadState(statePath string) error {
 	return file.Close()
 }
 
+func (s *Server) loadLatestState() error {
+	files, err := ioutil.ReadDir(filepath.Join(statesFolderName))
+	if err != nil {
+		return err
+	}
+	latestStateName := files[indexOfNewestFile(files)].Name()
+	if err := s.loadState(filepath.Join(statesFolderName, latestStateName)); err != nil {
+		return err
+	}
+	return err
+}
+
+func indexOfNewestFile(files []os.FileInfo) uint {
```

Comment:
We don't make use of the index, but of the name of the newest File so this func should just return the name.

Rather than looking at the time the files were modified, I would parse the name of the files and look at the timestamps we have written into them (it is possible that we are going to have other files in the same directory)